stdlib_test.go

```diff
@@ -251,11 +253,30 @@ func TestStdlibAdapterSubexps(t *testing.T) {
 			"msg":  ":.;<>_#{[]}\"\\",
 		},
 	} {
-		haveMap := subexps([]byte(input))
+		haveMap := subexps(StdlibAdapter{logRegexp: StdlibRegexpFull}, []byte(input))
 		for key, want := range wantMap {
 			if have := haveMap[key]; want != have {
 				t.Errorf("%q: %q: want %q, have %q", input, key, want, have)
 			}
 		}
```

Comment:
OK, I see. 

The point of this test is to ensure that the various fields of typical stdlib log messages are correctly extracted. Previously, a single regexp was responsible for that work, so this test didn't need to parameterize on that dimension. In effect, this test was just exercising that regexp, which is now known as StdlibRegexpFull. It only used subexps as a convenience.

This PR allows users to provide any regexp they want to parse stdlib log messages, and refactors subexps to parameterize on that arbitrary user input. This means subexps is no longer one-to-one with the StdlibRegexpFull regexp, which is what this test was actually testing. 

(Converting subexps from a free function to an unexported method on the adapter type also makes it more difficult to test, as is demonstrated here by the need to manually instantiate a StdlibAdapter with a logRegexp field.)

Summarizing: this test exercises regexp(s), not the subexp function. Let's refactor it, and expand it, to cover all of the regexps that are added by this PR.

What does that look like?

Based on the changes in the PR, each test case will probably need to define (a) the regexp that the adapter will use, (b) an input string of some stdlib log message, and (c) an expected output, containing individual fields that should be extracted, somehow. This last point is the major change. It no longer makes sense to test against the subexps method, so I think the best approach is probably to test against actual log output produced from the adapter.

For example, this test case

```go
		"2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello world": {
			"date": "2009/01/23",
			"time": "01:23:23.123123",
			"file": "/a/b/c/d.go:23",
			"msg":  "hello world",
		},
```

represents the implicit regexp of StdlibRegexpFull (which we should make explicit), an input string which is the map key, and some output which is currently modeled as a set of key-value pairs in a map, but that's only because that's how subexps worked. So let's change that expected output to be the e.g. logfmt output produced from the input parameters, which in this case I believe should be

```
date="2009/01/23" time="01:23:23.123123" file="/a/b/c/d.go:23" msg="hello world"
```

So the refactored version of this test case would look something like

```go
{
	regexp: StdlibRegexpFull,
	input:  `2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello world`,
	want:   `date="2009/01/23" time="01:23:23.123123" file="/a/b/c/d.go:23" msg="hello world"`,
}
```

And the loop body would look something like

```go
for _, testcase := range ... {
	var buf bytes.Buffer
	adapter := NewStdlibAdapter(NewLogfmtLogger(&buf), StdlibRegexp(testcase.regexp))
	fmt.Fprint(adapter, testcase.input)
	if want, have := testcase.want, buf.String(); want != have { 
		t.Errorf("%q: want %q, have %q", input, want, have)
	}
}
```

With this structure in place, we should then add test cases for each of the new exported regexps that are added by this PR.

Does this make sense?

Reply:
I test your approch, but it wont work. For example, the date and time subs from the regex function are converted to an ts field

```
=== RUN   TestStdlibAdapterSubexps
    stdlib_test.go:465: "2009/01/23 01:23:23.123123 /a/b/c/d.go:23: hello world": want "date=\"2009/01/23\" time=\"01:23:23.123123\" file=\"/a/b/c/d.go:23\" msg=\"hello world\"", have "ts=\"2009/01/23 01:23:23.123123\" caller=/a/b/c/d.go:23 msg=\"hello world\"\n"
--- FAIL: TestStdlibAdapterSubexps (0.00s)
```

From my point of view, I consider to remove the test again. The approch before was rejected and the advertised approch is already covered by other test cases. However I refactor the tests have a const pattern.